README.md

```diff
@@ -109,6 +109,43 @@ To get:
 ["a","b","c"]
 ```
 
+#### Safe iteration of YAML/JSON elements
+
+The `--as-json-lines` option will convert each top level element to a [JSON lines](https://jsonlines.org/) object.
+This allows you to safely process the results one line at a time with a simple line oriented reader.
+For example:
+
+```bash
+readarray -t dbs < <(clconf --pipe jsonpath '$[*]' --as-json-lines <<'EOF'
+---
+foodb:
+  host: foo.example.com
+  credentials:
+    username: foouser
+    password: foopass
+bardb:
+  host: bar.example.com
+  credentials:
+    username: baruser
+    password: barpass
+EOF
+)
+for db in "${dbs[@]}"; do
+  host="$(clconf --pipe getv /host <<<"${db}")"
+  user="$(clconf --pipe getv /credentials/username <<<"${db}")"
+  pass="$(clconf --pipe getv /credentials/password <<<"${db}")"
```

Comment:
Given the implementation wouldn't these be:

```suggestion
  host="$(clconf --pipe getv /value/host <<<"${db}")"
  user="$(clconf --pipe getv /value/credentials/username <<<"${db}")"
  pass="$(clconf --pipe getv /value/credentials/password <<<"${db}")"
```

May be worth making the example runnable so we can confirm its output? (e.g. swap in an `echo` for the `db_query`)

Reply:
Nope, this is using `jsonpath` which will result in a top level list.